### ISSUE TITLE
Maybe move message-storage settings out of persistent config block?

### ISSUE THREAD

**Issue by DanielOaks** [oragono/oragono#1128]
Looking through the config, it's a little unclear why these options are under the `persistent` block in the history config:

- `unregistered-channels`
- `registered-channels`
- `direct-messages`

It feels like storing history for unregistered channels, as well as setting the registered channels / direct message storage policy is a general history thing, rather than something to do with whether the persistent storage is active or not?

**Comment by slingamn** [comment]
These are actually specific to persistent history. For example:

1. `history.enabled: true`, `history.persistent.unregistered-channels: false`: unregistered channels have ephemeral history
2. `history.enabled: true`, `history.persistent.unregistered-channels: opt-out`: unregistered channels have persistent history by default, but the channel owner can switch to ephemeral or disable

Maybe the comments/documentation need to be clarified?

**Comment by DanielOaks** [comment]
Ah, so `false` means that it'll have history, it'll just be stored in the in-memory ring buffer rather than in the persistent store? Alright that totally makes sense then

**Comment by slingamn** [comment]
Yeah. It might make sense to clarify that, and maybe make it more clear what configurations are possible. (For example, it is not possible for the server operator to have history disabled entirely for unregistered channels, but allow either ephemeral or persistent history for registered channels.)
